lib/semantics/mod-file.cc

```diff
@@ -13,32 +13,45 @@
 // limitations under the License.
 
 #include "mod-file.h"
+#include "resolve-names.h"
 #include "scope.h"
 #include "semantics.h"
 #include "symbol.h"
 #include "../evaluate/traversal.h"
+#include "../parser/message.h"
 #include "../parser/parsing.h"
 #include <algorithm>
 #include <cerrno>
+#include <cstdio>
 #include <fstream>
 #include <ostream>
+#include <set>
+#include <string_view>
+#include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <unistd.h>
 #include <vector>
 
 namespace Fortran::semantics {
 
 using namespace parser::literals;
 
-// The initial characters of a file that identify it as a .mod file.
+// The first line of a file that identify it as a .mod file.
 // The first three bytes are a Unicode byte order mark that ensures
 // that the module file is decoded as UTF-8 even if source files
 // are using another encoding.
-static constexpr auto magic{"\xef\xbb\xbf!mod$ v1 sum:"};
+static constexpr struct {
+  const char *bom{"\xef\xbb\xbf"};
```

Comment:
These could be `static` members, and the lengths could then be calculated (or checked).

Reply:
Yes, much better that way.